### PR TITLE
Migrate to new Backend System

### PR DESCRIPTION
**Pull request by awanlin**
This PR migrates the backend to the new Backend System 🚀 

For now we are keeping the old backend files around until we have everything migrated, the explore plugin is the only hold over but that should be resolved soon as there is a PR open for that: https://github.com/backstage/backstage/pull/20551

Nice to have would be imports for the GraphQL plugin by Frontside

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     ]
   },
   "resolutions": {
+    "@backstage/backend-app-api": "0.5.7",
```
Comment (Rugvip) [inline package.json:32]: You should be able to remove this line and thing should still work

File: packages/backend/src/index.old.ts
Comment (Rugvip) [inline packages/backend/src/index.old.ts:1]: Naaah, delete this file and the test, we can find them in git if needed :>

**Review by Rugvip — APPROVED** [review]
Nice 😁, just some nits

File: packages/backend/src/index.old.ts
Comment (awanlin) [reply packages/backend/src/index.old.ts:1]: YOLO! Done ;)

**Pull request merged**